applications/FluidDynamicsApplication/custom_elements/surface_smoothing_element.cpp

```diff
@@ -0,0 +1,670 @@
+}
+
+/**
+ * creates a new element pointer and clones the previous element data
+ * @param NewId: the ID of the new element
+ * @param ThisNodes: the nodes of the new element
+ * @param pProperties: the properties assigned to the new element
+ * @return a Pointer to the new element
+ */
+Element::Pointer SurfaceSmoothingElement::Clone(IndexType NewId, NodesArrayType const& ThisNodes) const
+{
+    KRATOS_TRY
+    return Kratos::make_intrusive<SurfaceSmoothingElement>(NewId, GetGeometry().Create(ThisNodes), pGetProperties());
+    KRATOS_CATCH("");
+}
+
+/**
+ * this determines the elemental equation ID vector for all elemental
+ * DOFs
+ * @param rResult: the elemental equation ID vector
+ * @param rCurrentProcessInfo: the current process info instance
+ */
+void SurfaceSmoothingElement::EquationIdVector(EquationIdVectorType& rResult, ProcessInfo& CurrentProcessInfo)
+{
+    // num_dof = num_nodes
+    if (rResult.size() != num_nodes){
+        rResult.resize(num_nodes, false);
+    }
+
+    for(unsigned int i=0; i<num_nodes; i++){
+        rResult[i]  =  this->GetGeometry()[i].GetDof(DISTANCE).EquationId();
+    }
+}
+
+/**
+ * determines the elemental list of DOFs
+ * @param ElementalDofList: the list of DOFs
+ * @param rCurrentProcessInfo: the current process info instance
+ */
+void SurfaceSmoothingElement::GetDofList(DofsVectorType& rElementalDofList, ProcessInfo& CurrentProcessInfo)
+{
+    // Here, num_dof = num_nodes
+    if (rElementalDofList.size() != num_nodes){
+        rElementalDofList.resize(num_nodes);
+    }
+
+    for(unsigned int i=0; i<num_nodes; i++){
+        rElementalDofList[i] = this->GetGeometry()[i].pGetDof(DISTANCE);
+    }
+}
+
+/**
+ * ELEMENTS inherited from this class have to implement next
+ * CalculateLocalSystem, CalculateLeftHandSide and CalculateRightHandSide methods
+ * they can be managed internally with a private method to do the same calculations
+ * only once: MANDATORY
+ */
+
+/**
+ * this is called during the assembling process in order
+ * to calculate all elemental contributions to the global system
+ * matrix and the right hand side
+ * @param rLeftHandSideMatrix: the elemental left hand side matrix
+ * @param rRightHandSideVector: the elemental right hand side
+ * @param rCurrentProcessInfo: the current process info instance
+ */
+void SurfaceSmoothingElement::CalculateLocalSystem(
+    MatrixType& rLeftHandSideMatrix,
+    VectorType& rRightHandSideVector,
+    ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    const double dt = rCurrentProcessInfo.GetValue(DELTA_TIME);
+    const double he = ElementSizeCalculator<3,4>::AverageElementSize(GetGeometry());
+    const double epsilon = 1.0e2*dt*he*he;
+
+    BoundedMatrix<double,num_nodes,num_dim> DN_DX;  // Gradients matrix
+    array_1d<double,num_nodes> N; // dimension = number of nodes . Position of the gauss point
+
+    array_1d<double,num_nodes> PHIdof; //dimension = number of DOFs, needed since we are using a residualbased approach
```

Comment:
please check the [Style Guide](https://github.com/KratosMultiphysics/Kratos/wiki/Style-Guide#coding-convention) for naming conventions

Reply:
The style is now followed for phi_dof, etc.